packages/arb-ts/integration_test/arb-bridge.test.ts

```diff
@@ -385,14 +386,172 @@ describe('ERC20', () => {
+        )
+      )[0]
+
+      expect(eventData).to.exist
+
+      const { seqNum } = eventData
+
+      const l2RetriableHash = await bridge.calculateL2RetryableTransactionHash(
+        seqNum
+      )
+
+      const retriableReceipt = await arbProvider.waitForTransaction(
+        l2RetriableHash
+      )
+
+      expect(retriableReceipt.status).to.equal(1)
+
+      wait()
+      const l2AddressHopefully = await bridge.arbTokenBridge.customToken(
+        l1CustomToken.address
+      )
+      expect(l2AddressHopefully).to.equal(l2CustomToken.address)
+    })
+  } else {
+    l1CustomToken = TestCustomTokenL1__factory.connect(
+      existantCustomTokenL1,
+      preFundedWallet
+    )
+    l2CustomToken = TestArbCustomToken__factory.connect(
+      existantCustomTokenL2,
+      l2TestWallet
+    )
+    prettyLog(
+      `Connected to pre-deployed custom token addresses. L1:${l1CustomToken.address} L2: ${l2CustomToken.address}`
+    )
+  }
+
+  it('setup: mint some custom token and send to test address', async () => {
+    const mintRes = await l1CustomToken.mint()
+    const mintRec = await mintRes.wait()
+    const bal = await l1CustomToken.balanceOf(preFundedWallet.address)
+    expect(bal.gt(BigNumber.from(40000000))).to.be.true
+
+    erc20Address = l1CustomToken.address
+    const res = await l1CustomToken.transfer(
+      l1TestWallet.address,
+      BigNumber.from(200)
+    )
+    const rec = await res.wait()
+    const data = await bridge.getAndUpdateL1TokenData(erc20Address)
+    const testWalletBal = data.ERC20 && data.ERC20.balance
+    expect(testWalletBal && testWalletBal.eq(BigNumber.from(200))).to.be.true
+  })
+
+  it('approve token for bridge contract', async () => {
+    const approveRes = await bridge.approveToken(l1CustomToken.address)
+    const approveRec = await approveRes.wait()
+    expect(approveRec.status).to.equal(1)
+
+    const data = await bridge.getAndUpdateL1TokenData(l1CustomToken.address)
+    const allowed = data.ERC20 && data.ERC20.allowed
+    expect(allowed).to.be.true
+  })
+  it('deposits custom token', async () => {
+    const despositRes = await bridge.depositAsCustomToken(
+      l1CustomToken.address,
+      tokenDepositAmmount,
+      BigNumber.from(10000000000000),
+      BigNumber.from(0),
+      '',
+      { gasLimit: 210000, gasPrice: l1gasPrice }
+    )
+
+    const depositRec = await despositRes.wait()
+    expect(depositRec.status).to.equal(1)
+    await wait(10000)
+
+    const tokenDepositData = (
+      await bridge.getDepositTokenEventData(depositRec)
+    )[0] as DepositTokenEventResult
+
+    const seqNum = tokenDepositData.seqNum
+
+    const l2RetriableHash = await bridge.calculateL2RetryableTransactionHash(
+      seqNum
+    )
+
+    const retriableReceipt = await arbProvider.waitForTransaction(
+      l2RetriableHash
+    )
+
+    expect(retriableReceipt.status).to.equal(1)
+  })
+
+  it('wallet has expected balance after custom token deposit', async () => {
```

Comment:
Should this not be part of the previous `deposit` test case?

Reply:
no strong reason either way; i lean toward more, smaller cases if possible for more granular error logging